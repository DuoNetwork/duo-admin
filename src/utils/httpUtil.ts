import request from 'request';

class HttpUtil {
	public get(url: string, headerOthers?: object): Promise<any> {
		return new Promise((resolve, reject) =>
			request(
				{
					url,
					headers: Object.assign(
						{
							'user-agent': 'node.js'
						},
						headerOthers || {}
					)
				},
				(error, res, body) => {
					if (error) reject(error);
					else if (res.statusCode === 200) resolve(body);
					else reject('Error status ' + res.statusCode + ' ' + res.statusMessage);
				}
			)
		);
	}

	public postJson(url: string, json: object): Promise<object> {
		return new Promise((resolve, reject) =>
			request(
				{
					url,
					method: 'POST',
					headers: {
						'Content-Type': 'application/json'
					},
					json
				},
				(error, res, body) => {
					if (error) reject(error);
					else if (res.statusCode === 200) resolve(body);
					else reject('Error status ' + res.statusCode + ' ' + res.statusMessage);
				}
			)
		);
	}
}

const httpUtil = new HttpUtil();
export default httpUtil;
