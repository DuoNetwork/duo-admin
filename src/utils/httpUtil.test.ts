import httpUtil from './httpUtil';

let error = '';
const res = { statusCode: 200, statusMessage: 'statusMessage' };
const body = 'body';
jest.mock('request', () =>
	jest.fn((option: any, cb: any) => {
		if (option) cb(error, res, body);
	})
);

test('get', async () => {
	expect(await httpUtil.get('url')).toBe('body');
});

test('get error', async () => {
	error = 'error';
	try {
		await httpUtil.get('url');
	} catch (error) {
		expect(error).toBe('error');
	}
});

test('get not 200', async () => {
	error = '';
	res.statusCode = 201;
	try {
		await httpUtil.get('url');
	} catch (error) {
		expect(error).toMatchSnapshot();
	}
});

test('post', async () => {
	res.statusCode = 200;
	expect(await httpUtil.postJson('url', {})).toBe('body');
});

test('post error', async () => {
	error = 'error';
	try {
		await httpUtil.postJson('url', {});
	} catch (error) {
		expect(error).toBe('error');
	}
});

test('post not 200', async () => {
	error = '';
	res.statusCode = 201;
	try {
		await httpUtil.postJson('url', {});
	} catch (error) {
		expect(error).toMatchSnapshot();
	}
});
