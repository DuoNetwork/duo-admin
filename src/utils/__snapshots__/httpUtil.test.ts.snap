// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get not 200 1`] = `"Error status 201 statusMessage"`;

exports[`post not 200 1`] = `"Error status 201 statusMessage"`;
